### ISSUE TITLE
jobs: during version upgrade long running transactions can fail with:  relation \"system.job_info\" does not exist

### ISSUE THREAD

**Issue by fqazi** [cockroachdb/cockroach#99241]
We modified the acceptance/version-upgrade to start doing DDL statements that can end up creating/ updating jobs, while running this workload we encountered the following error:

```
"message": "***UNEXPECTED COMMIT ERROR; Received an unexpected commit error: ERROR: write-job-info-delete: relation \"system.job_info\" does not exist (SQLSTATE 42P01)",
```

This is likely happening because we had a long-running transaction attempt to update the jobs table after the cluster version is bumped. Since the query the active version is in memory, it doesn't take into account if the transaction can actually *see* this table:
https://github.com/cockroachdb/cockroach/blob/master/pkg/jobs/registry.go#L740

@renatolabs Uploaded logs from the bad run where we saw this happen: 
https://github.com/cockroachdb/cockroach/files/11041863/99162_artifacts.zip

Jira issue: CRDB-25779

**Comment by knz** [comment]
fixed by #100523

**Comment by knz** [comment]
oh actually let's keep this open until the backport PR is merged. #100538

**Comment by knz** [comment]
fixed by #100538
